cmake/module/AddBoostIfNeeded.cmake

```diff
@@ -55,22 +55,16 @@ function(add_boost_if_needed)
   endif()
 
   if(BUILD_TESTS)
-    include(CheckCXXSourceCompiles)
-    check_cxx_source_compiles("
-      #define BOOST_TEST_MAIN
-      #include <boost/test/included/unit_test.hpp>
-      " HAVE_BOOST_INCLUDED_UNIT_TEST_H
-    )
+    # Some package managers, such as vcpkg, install Boost headers individually
+    # for each package. Therefore, even after a successful `FindBoost` call, we
+    # explicitly check for the Boost.Test headers.
+    list(APPEND CMAKE_REQUIRED_DEFINITIONS -DBOOST_TEST_NO_MAIN)
+    include(CheckIncludeFileCXX)
+    check_include_file_cxx(boost/test/included/unit_test.hpp HAVE_BOOST_INCLUDED_UNIT_TEST_H)
     if(NOT HAVE_BOOST_INCLUDED_UNIT_TEST_H)
       message(FATAL_ERROR "Building test_bitcoin executable requested but boost/test/included/unit_test.hpp header not available.")
     endif()
-
-    check_cxx_source_compiles("
-      #define BOOST_TEST_MAIN
-      #include <boost/test/included/unit_test.hpp>
-      #include <boost/test/unit_test.hpp>
-      " HAVE_BOOST_UNIT_TEST_H
-    )
+    check_include_file_cxx(boost/test/unit_test.hpp HAVE_BOOST_UNIT_TEST_H)
```

Comment:
I think I might be missing something, but this is still checking two headers.

Reply:
Thanks! Reworked.

Only one Boost.Test header is being checked now.